### ISSUE TITLE
Unexpected Installation of Beta Release Despite "Include beta releases" Being Unchecked

### ISSUE THREAD

**Issue by ivy** [transmission/transmission#7312]
### What is the issue?

Hey folks! Transmission prompted me to install an update just now when starting the app. Version `4.1.0-beta.1 (c3a8106077)` was installed, even though the "Include beta releases" option is unchecked in the "Check for update:" section of the settings.

**Steps to Reproduce:**  
1. Open Transmission client.  
2. Ensure the "Include beta releases" option is unchecked in Preferences under "Check for update."  
3. Accept the prompt to install an available update.  
4. Verify the installed version.  

**Expected Behavior:**  
When "Include beta releases" is unchecked, only stable releases should be installed.  

**Actual Behavior:**  
A beta release (`4.1.0-beta.1`) was installed despite the "Include beta releases" option being unchecked.  

**Environment:**  
- **Transmission version:** 4.1.0-beta.1 (c3a8106077) — Prior to updating, I'm not sure what release I was running but it's been about a week since I last ran Transmission.
- **Operating system:** macOS  
- **System details:**  
  ```
  $ uname -a
  Darwin Ivys-MacBook-Air.local 24.1.0 Darwin Kernel Version 24.1.0: Thu Oct 10 21:05:14 PDT 2024; root:xnu-11215.41.3~2/RELEASE_ARM64_T8103 arm64
  ```
- **Settings:** "Include beta releases" is unchecked in Preferences.

**Additional Information:**  

<details><summary>Screenshots</summary>

<img width="752" alt="Screenshot 2024-12-18 at 5 17 11 PM" src="https://github.com/user-attachments/assets/20f556ed-e998-4085-b9ce-a8f40a0e135b" />

<img width="650" alt="Screenshot 2024-12-18 at 5 17 04 PM" src="https://github.com/user-attachments/assets/0aeb1887-2896-4a37-b222-0c3818ae09b8" />

</details>

### Which application of Transmission?

macOS app

### Which version of Transmission?

4.1.0-beta.1 (c3a8106077)

**Comment by livings124** [comment]
Thanks for the report. What version were you updating from? I have tested 4.0.6 and the checkbox was respected, but earlier versions might not.

**Comment by Coeur** [comment]
Author wrote:
>Prior to updating, I'm not sure what release I was running

**Comment by ivy** [comment]
Yeah, what @.Coeur mentioned. If there's a way to figure out what the previously installed version was, I could take another look!

**Comment by ivy** [comment]
Okay, checking back in, I can confirm it was 4.0.5 that has this bug:
<img width="752" alt="Screenshot 2024-12-19 at 7 50 01 PM" src="https://github.com/user-attachments/assets/f1545e85-f7a7-4213-a6aa-7e0feb4afccb" />

(☝️ Seen when clicking "Check For Update..." with the settings I mentioned before.)

**Comment by Coeur** [comment]
My version fix was done in #6623 which landed in March 2024, meaning with 4.0.6 and newer.
Transmission 4.0.5 was in December 2023.
I'm sorry that we didn't get an ideal 4.0.6 release and that 4.0.7 never happened to address that.

**Comment by ivy** [comment]
Aww no hard feelings! Well feel free to close this issue. It sounds like you fixed it!

Unfortunately, the beta and 4.0.6 aren't allowed by some of my trackers. 😭 Bugs happen and I imagine you all dealt with some angry users. Which is a shame! Transmission is probably my most used app after my web browser. 🙃 

Thanks for looking into my issue!
